### PR TITLE
refactor: rename symbol `StructType` to `Struct` and ttype `Struct` to `StructType`

### PR DESCRIPTION
**Pull request by kmr-srbh**
Rename symbol `StructType` to `Struct` and ttype `Struct` to `StructType`.

### CODE REVIEW

File: src/libasr/ASR.asdl [hunk 1]
```diff
@@ -150,8 +150,8 @@ expr
     | ArrayReshape(expr array, expr shape, ttype type, expr? value)
     | ArrayBroadcast(expr array, expr shape, ttype type, expr? value)
     | BitCast(expr source, expr mold, expr? size, ttype type, expr? value)
-    | StructInstanceMember(expr v, symbol m, ttype type, expr? value)
-    | StructStaticMember(expr v, symbol m, ttype type, expr? value)
+    | StructTypeInstanceMember(expr v, symbol m, ttype type, expr? value)
+    | StructTypeStaticMember(expr v, symbol m, ttype type, expr? value)
```
Comment (certik) [inline src/libasr/ASR.asdl:154]: This should stay with the old name.

**Comment by certik** [comment]
This is good. I think for now, let's do `ttype StructType` and `symbol Struct` and all operations on the given type are just `StructDoSomething`.

For functions, we have `ttype FunctionType`, and `symbol Function` and operations on functions would be `FunctionCall`.

Finally, maybe later we'll do `ttype IntegerType`, the corresponding symbol is just `Variable` (already the current design), and operations on integers are `IntegerDoSomething` (already the current design).

File: src/libasr/ASR.asdl (hunk 1 above)
Comment (kmr-srbh) [reply src/libasr/ASR.asdl:154]: Pushing these changes.

**Comment by Pranavchiku** [comment]
Well if this is not urgent and if I by chance get #4247 working by tonight, I'll request you to let that PR merge first

**Comment by certik** [comment]
I think the ASR.asdl change is now correct.

**Comment by Pranavchiku** [comment]
I merged #4256, thanks for waiting! We can proceed with this PR.

**Comment by kmr-srbh** [comment]
@certik can this PR be merged?

**Comment by Thirumalai-Shaktivel** [comment]
@kmr-srbh, can you please submit a PR with the same changes to LPython?

**Comment by certik** [comment]
@Thirumalai-Shaktivel he is, I already asked him. :)

**Pull request merged**